src/string/trie.hpp

```diff
@@ -3,47 +3,40 @@
 using namespace std;
 
 struct Trie {
-  private:
     static constexpr int C_SIZE = 26;    // C_SIZE  : 文字の種類数
     static constexpr int C_BEGIN = 'a';  // C_BEGIN : 開始文字
-    int root = 0;
+    static constexpr int ROOT = 0;
     struct Node {
         int child[C_SIZE];  // 子ノードの番号, 存在しなければ-1
         vector<int> ids;    // そのノードが終端である文字列のIDリスト
         Node() { fill(child, child + C_SIZE, -1); }
     };
-
-  public:
     vector<Node> nodes;
     int cnt = 0;  // 追加した文字列の個数
     Trie() : nodes(1) {}
     // nodes[idx]から文字cで遷移したときの頂点のindex
-    int next_index(int idx, char c) {
-        return nodes[idx].child[c - C_BEGIN];
-    }
+    int next(int idx, char c) { return nodes[idx].child[c - C_BEGIN]; }
     // 文字列の追加
-    void insert(const string str) {
-        int now = root;
-        for (auto c : str) {
+    int insert(const string& s) {
+        int now = ROOT;
```

Comment:
```suggestion
        int idx = ROOT;
```
`find()` の実装に合わせて変数名を `idx` にするのはどうでしょうか

Reply:
Aho-Corasick法では関数の引数に `idx` が使われるので、対応を考えると `now` の方が良さそうですね。
`find()` の方を `now` にすると良いかもしれません。

Reply:
確かに他の関数では `now` をしようしているのでこちらに統一しようと思います

Reply:
[f9f29e4](https://github.com/YNUCPC/ynu-icpc-library/pull/80/commits/f9f29e4d0c8f79bfdef8f3725af5ab88a10168ac) で対応しました